core/src/proof/path_proof.rs

```diff
@@ -324,8 +331,9 @@ pub fn verify_update<H: NodeHasher>(
         pending_siblings.push((cur_node, end_layer));
     }
 
-    // The last `path` iterates up to layer 0, so this always stores the root if paths nonempty.
-    Ok(pending_siblings.pop().map(|n| n.0).unwrap_or(TERMINATOR))
+    // UNWRAP: If `paths` is not empty this can never be `None` since `pending_siblings` is
+    // unconditionally appended to.
+    Ok(pending_siblings.pop().map(|n| n.0).unwrap())
```

Comment:
i feel like it's slightly cleaner to just do `unwrap_or(prev_root)` here, but patch is fine as-is

Reply:
I've considered this but opted-in for the solution presented here because it is more explicit and slightly less error prone. You can see how this promise of returning [] → prev_root is fulfied right away, instead of scrolling 1.5 screens worth of code.